tests/ui/from_iter_instead_of_collect.rs

```diff
@@ -0,0 +1,15 @@
+#![warn(clippy::from_iter_instead_of_collect)]
+
+use std::collections::HashMap;
+use std::iter::FromIterator;
+
+fn main() {
+    {
+        let iter_expr = std::iter::repeat(5).take(5);
+
+        Vec::from_iter(iter_expr);
+        HashMap::<usize, &i8>::from_iter(vec![5, 5, 5, 5].iter().enumerate());
+        //let v: Vec<i32> = iter_expr.collect();
+        let a: Vec<i32> = Vec::new();
```

Comment:
Can this be removed or do you have some more tests in mind you want to use this for?